starcompactor/machine_event_dump.py

```diff
@@ -77,32 +79,34 @@ def refine_machine_events(machine_events):
         if event_type == 'ENABLE': is_enabled = True
         if event_type == 'DISABLE': is_enabled = False
     
-    return machine_events  
+    return machine_events
 
-def mask_and_derive(machine_events, masker, epoch):
+def mask_and_derive(machine_events, masker, epoch, rack_property_name):
     traces = []
-    # ordered mask rack
-    ordered_racks = sorted(set([machine_events[k]['RACK'] for k in machine_events.keys()]))
     for key in machine_events.keys():
         trace = machine_events[key].copy()
         trace['EVENT_TIME'] = key[0]
         trace['HOST_NAME (PHYSICAL)'] = key[1]
         trace['EVENT'] = key[2]
         trace = trans.mask_fields(trace, TRACE_TYPE, masker)
         trace = trans.machine_event_times(trace, epoch)
-        trace = trans.ordered_mask(trace, 'RACK', ordered_racks) 
+        if rack_property_name:
+            trace = trans.ordered_mask(trace, rack_property_name, sorted(set([machine_events[k][rack_property_name] for k in machine_events.keys() if rack_property_name in machine_events[k]]))) 
         traces.append(trace) 
     
     return traces
 
 def get_machine_event_with_packed_args(args):
     try:
-        return machine.get_machine_event(*args)
+        return machine.get_machine_event(**args)
```

Comment:
Not sure this should be `**` - you want to unpack the positional args, right? `args` isn't a dict, I think it's a list here.

Reply:
I switched it to dict https://github.com/ChameleonCloud/starcompactor/blob/chi/starcompactor/machine_event_dump.py#L159-L162.

Reply:
Ah OK, I misunderstood the `map` syntax